### PR TITLE
fix the override reconcile timeout test (#320)

### PR DESCRIPTION
**Pull request by sdowell**
TestOverrideReconcileTimeout is failing on Autopilot cluster, add logic to scale up the Autopilot cluster before creating the pod for test.

Pull in resourcegroup package and update klog version from v2.60.1 to v2.70.1

### CODE REVIEW

**Comment by sdowell** [comment]
I believe this should fix the test failure observed [here](https://testgrid.k8s.io/googleoss-kpt-config-sync-release-multi-repo-2#autopilot-stable). PTAL
/hold
/cc @xinnywinne 
/cc @tiffanny29631 
/cc @nan-yu

**Review by xinnywinne — APPROVED** [review]
/lgtm

File: e2e/testcases/override_reconcile_timeout_test.go [hunk 1]
```diff
@@ -36,10 +38,31 @@ func TestOverrideReconcileTimeout(t *testing.T) {
 	nt := nomostest.New(t, nomostesting.OverrideAPI, ntopts.Unstructured)
 	rootSync := fake.RootSyncObjectV1Beta1(configsync.RootSyncName)
 
-	// Override reconcileTimeout to a short time 30s, pod will never be ready.
-	// Only actuation should succeed, reconcile should time out.
-	nt.MustMergePatch(rootSync, `{"spec": {"override": {"reconcileTimeout": "30s"}}}`)
+	// Override reconcileTimeout to a short time 30s, only actuation should succeed, reconcile should time out.
+	if err := nt.Client.Patch(nt.Context, rootSync, client.RawPatch(types.MergePatchType,
```
[the remark this answers is not in the document]
Comment (sdowell) [reply e2e/testcases/override_reconcile_timeout_test.go:42]: This is a cherry pick, so I don't think we should make any changes. If we want to change this we can make a new commit to main IMO

**Review by nan-yu — APPROVED** [review]
/lgtm

**Pull request merged**